packages/gestalt/src/contexts/DefaultLabelProvider.js

```diff
@@ -0,0 +1,66 @@
+// @flow strict
+import { type Context, type Node, createContext, useContext } from 'react';
+
+/**
+ * To add new labels:
+ * - Create a type for the component's labels (these types need to be flat, *not* nested)
+ * - Add component labels type to DefaultLabelContextType keyed by component name
+ * - Add fallback labels to fallbackLabels below
+ */
+
+export type DefaultLabelContextType = {|
+  ComboBox: {|
+    accessibilityClearButtonLabel: string,
+  |},
+  TextField: {|
+    accessibilityHidePasswordLabel: string,
+    accessibilityShowPasswordLabel: string,
+  |},
+|};
+
+export const fallbackLabels: DefaultLabelContextType = {
+  ComboBox: {
+    accessibilityClearButtonLabel: 'Clear input',
+  },
+  TextField: {
+    accessibilityHidePasswordLabel: 'Hide password',
+    accessibilityShowPasswordLabel: 'Show password',
+  },
+};
+
+const DefaultLabelContext: Context<?DefaultLabelContextType> =
+  createContext<?DefaultLabelContextType>(null);
+
+type Props = {|
+  children: Node,
+  /**
+   * An object describing the default strings to be used by supported component labels throughout your app. If your app supports i18n, don't forget to translate your strings!
+   *
+   * Note that all supported labels for all supported components must be provided if using this Provider. Omit this prop to use default (English) strings for supported labels.
+   *
+   * See [the source code](https://github.com/pinterest/gestalt/blob/master/packages/gestalt/src/contexts/DefaultLabelProvider.js) for the specific shape of this object.
+   */
+  labels?: null | DefaultLabelContextType,
```

Comment:
Strangely, when using `?DefaultLabelContextType` here, the type is shown in full on the docs page. Using `null | DefaultLabelContextType` just shows that literal union. No idea why. 🤔 

This is actually an advantage here, though. This type is already complicated and will be very gnarly in the future, and the lack of formatting makes it look like crap on the docs pages. Given that this Provider is more of an "admin" feature, I'm fine with pointing people to the source code to see the actual object shape. It also means we maintain a single source of truth, which is very important with a living object like this.